app/src/main/java/com/sun_asterisk/moviedb_49/data/source/MovieDataSource.kt

```diff
@@ -0,0 +1,18 @@
+package com.sun_asterisk.moviedb_49.data.source
+
+import com.sun_asterisk.moviedb_49.data.model.Movie
+import com.sun_asterisk.moviedb_49.data.source.remote.OnFetchDataJsonListener
+
+class MovieDataSource private constructor(private val remote: MovieDBDataSource.Remote) {
+
+    fun getData(typeMovie: String, page: Int, listener: OnFetchDataJsonListener<Movie>) {
+        remote.getData(typeMovie, page, listener)
+    }
+
+    companion object {
+        private var instance: MovieDataSource? = null
+
+        fun getsInstance(remote: MovieDBDataSource.Remote): MovieDataSource =
+            instance ?: MovieDataSource(remote)
```

Comment:
Lần trước suggest thiếu 
```suggestion
        fun getInstance(remote: MovieDBDataSource.Remote): MovieDataSource =
            instance ?: MovieDataSource(remote).also { instance = it }
```